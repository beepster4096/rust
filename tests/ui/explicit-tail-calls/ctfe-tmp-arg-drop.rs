#![feature(explicit_tail_calls, const_trait_impl, const_mut_refs)]

pub const fn test(_: &View) {
    const fn takes_view(_: &View) {}

    become takes_view(HasDrop.as_view());
    //~^ error: temporary value dropped while borrowed
}

struct HasDrop;
struct View;

impl HasDrop {
    const fn as_view(&self) -> &View {
        &View
    }
}

impl const Drop for HasDrop {
    fn drop(&mut self) {}
}

fn main() {}
