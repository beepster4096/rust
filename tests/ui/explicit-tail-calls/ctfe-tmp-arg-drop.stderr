error[E0716]: temporary value dropped while borrowed
  --> $DIR/ctfe-tmp-arg-drop.rs:6:23
   |
LL |     become takes_view(HasDrop.as_view());
   |     ------------------^^^^^^^------------ temporary value is freed at the end of this statement
   |     |                 |
   |     |                 creates a temporary value which is freed while still in use
   |     borrow later used here
   |
help: consider using a `let` binding to create a longer lived value
   |
LL ~     let binding = HasDrop;
LL ~     become takes_view(binding.as_view());
   |

error: aborting due to previous error

For more information about this error, try `rustc --explain E0716`.
